includes/Modules/Analytics.php

```diff
@@ -1299,15 +1299,33 @@ protected function has_access_to_property( $property_id ) {
 	 * @return WP_Error WordPress error object.
 	 */
 	protected function exception_to_error( Exception $e, $datapoint ) {
+		$cache_ttl = false;
+
 		if ( 'report' === $datapoint && $e instanceof Google_Service_Exception ) {
 			$errors = $e->getErrors();
 			// If error is because of AdSense metric being requested, set adsenseLinked to false.
-			if ( isset( $errors[0]['message'] ) && $this->is_adsense_metric( substr( $errors[0]['message'], strlen( 'Restricted metric(s): ' ) ) ) ) {
-				$this->get_settings()->merge( array( 'adsenseLinked' => false ) );
+			if ( isset( $errors[0]['message'] ) ) {
+				if ( $this->is_adsense_metric( substr( $errors[0]['message'], strlen( 'Restricted metric(s): ' ) ) ) ) {
+					$this->get_settings()->merge( array( 'adsenseLinked' => false ) );
+				}
+
+				if ( preg_match( '#^Restricted metric\(s\)\:#im', $errors[0]['message'] ) ) {
+					$cache_ttl = ( 10 * MINUTE_IN_SECONDS );
+				}
 			}
 		}
 
-		return parent::exception_to_error( $e, $datapoint );
+		$error = parent::exception_to_error( $e, $datapoint );
+
+		if ( $cache_ttl && is_wp_error( $error ) ) {
+			$error->add_data(
+				array(
+					'cacheTTL' => $cache_ttl,
+				)
+			);
```

Comment:
This unconditionally overrides any existing `data` that is already present in the `$error` (e.g. the HTTP response code in `status` or the `errorReason`). We'll need to change it to manually alter the `$error->error_data` property if it already has content (modifying properties directly isn't great, but that's all that WordPress core allows here).